src/backend/base/langflow/api/v1/flows.py

```diff
@@ -291,27 +293,30 @@ async def update_flow(
             default_folder = (await session.exec(select(Folder).where(Folder.name == DEFAULT_FOLDER_NAME))).first()
             if default_folder:
                 db_flow.folder_id = default_folder.id
+
         session.add(db_flow)
         await session.commit()
         await session.refresh(db_flow)
+
+        return db_flow
+
     except Exception as e:
-        # If it is a validation error, return the error message
+        await session.rollback()
```

Comment:
I don't think this is a good idea. This should be in a centralized place just like we do with `session_scope`.

Something like this (in https://github.com/langflow-ai/langflow/blob/57d562b3a7452f1ec72c59a5761aee46a451d5dc/src/backend/base/langflow/services/database/service.py):
```python
    @asynccontextmanager
    async def with_session(self):
        async with AsyncSession(self.engine, expire_on_commit=False) as session:
            try:
                yield session
                if session.is_active:
                    await session.commit()
            except Exception:
                logger.exception("An error occurred during the session scope.")
                await session.rollback()
                raise
```